### DISCUSSION TITLE
Use markdown to display `output:///XXX`

### DISCUSSION THREAD

**Discussion by Freed-Wu** [neoclide/coc.nvim#4923] [Ideas]
Can we use markdown to display the result of `output:///XXX`?

For example, `output://languageserver.dot`:

```
[Trace - 13:02:32.968] Sending request 'initialize - (0)'.
Params: {
    "processId": 362866,
    "rootPath": "/home/wzy/Desktop",
    "rootUri": "file:///home/wzy/Desktop",
    "capabilities": {
      "...": "..."
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "...": "..."
        }
    ],
    "locale": "en_US",
    "clientInfo": {
        "name": "coc.nvim",
        "version": "0.0.82"
    }
}


[Trace - 13:02:33.279] Received response 'initialize - (0)' in 310ms.
Result: {
    "capabilities": {
        "textDocumentSync": 1,
        "completionProvider": {
            "triggerCharacters": [
                "="
            ],
            "resolveProvider": false
        },
        "hoverProvider": true,
        "referencesProvider": true,
        "definitionProvider": true,
        "renameProvider": true,
        "codeActionProvider": true,
        "executeCommandProvider": {
            "commands": [
                "DOT.changeEdgeOp",
                "DOT.convertGraphType",
                "DOT.consolidateDescendants",
                "DOT.removeSemicolons"
            ]
        },
        "colorProvider": true
    }
}


[Trace - 13:02:33.280] Sending notification 'initialized'.
Params: {}


[Trace - 13:02:33.292] Sending notification 'workspace/didChangeConfiguration'.
Params: {
    "settings": {}
}
```

There are many json without syntax highlight, which is hard to read.

We can improve it as:

````
## [Trace - 13:02:32.968] Sending request 'initialize - (0)'.

Params:

```json
{
    "processId": 362866,
    "rootPath": "/home/wzy/Desktop",
    "rootUri": "file:///home/wzy/Desktop",
    "capabilities": {
      "...": "..."
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "...": "..."
        }
    ],
    "locale": "en_US",
    "clientInfo": {
        "name": "coc.nvim",
        "version": "0.0.82"
    }
}
```


## [Trace - 13:02:33.279] Received response 'initialize - (0)' in 310ms.

Result:

```json
{
    "capabilities": {
        "textDocumentSync": 1,
        "completionProvider": {
            "triggerCharacters": [
                "="
            ],
            "resolveProvider": false
        },
        "hoverProvider": true,
        "referencesProvider": true,
        "definitionProvider": true,
        "renameProvider": true,
        "codeActionProvider": true,
        "executeCommandProvider": {
            "commands": [
                "DOT.changeEdgeOp",
                "DOT.convertGraphType",
                "DOT.consolidateDescendants",
                "DOT.removeSemicolons"
            ]
        },
        "colorProvider": true
    }
}
```


## [Trace - 13:02:33.280] Sending notification 'initialized'.

Params: 

```json
{}
```


## [Trace - 13:02:33.292] Sending notification 'workspace/didChangeConfiguration'.

Params:

```json
{
    "settings": {}
}
```
````

The result is like:

> ## [Trace - 13:02:32.968] Sending request 'initialize - (0)'.
> 
> Params:
> 
> ```json
> {
>     "processId": 362866,
>     "rootPath": "/home/wzy/Desktop",
>     "rootUri": "file:///home/wzy/Desktop",
>     "capabilities": {
>       "...": "..."
>     },
>     "trace": "verbose",
>     "workspaceFolders": [
>         {
>             "...": "..."
>         }
>     ],
>     "locale": "en_US",
>     "clientInfo": {
>         "name": "coc.nvim",
>         "version": "0.0.82"
>     }
> }
> ```
> 
> 
> ## [Trace - 13:02:33.279] Received response 'initialize - (0)' in 310ms.
> 
> Result:
> 
> ```json
> {
>     "capabilities": {
>         "textDocumentSync": 1,
>         "completionProvider": {
>             "triggerCharacters": [
>                 "="
>             ],
>             "resolveProvider": false
>         },
>         "hoverProvider": true,
>         "referencesProvider": true,
>         "definitionProvider": true,
>         "renameProvider": true,
>         "codeActionProvider": true,
>         "executeCommandProvider": {
>             "commands": [
>                 "DOT.changeEdgeOp",
>                 "DOT.convertGraphType",
>                 "DOT.consolidateDescendants",
>                 "DOT.removeSemicolons"
>             ]
>         },
>         "colorProvider": true
>     }
> }
> ```
> 
> 
> ## [Trace - 13:02:33.280] Sending notification 'initialized'.
> 
> Params: 
> 
> ```json
> {}
> ```
> 
> 
> ## [Trace - 13:02:33.292] Sending notification 'workspace/didChangeConfiguration'.
> 
> Params:
> 
> ```json
> {
>     "settings": {}
> }
> ```

It will be better, right?

**Comment by fannheyward** [comment]
I always use https://lsp-log-parser.netlify.app to see the trace logs, run locally with https://github.com/rchl/lsp-log-parser/pull/255
